packages/flutter_tools/test/hot_test.dart

```diff
@@ -83,4 +88,49 @@ void main() {
       }), true);
     });
   });
+
+  group('hotRestart', () {
+    final List<FlutterDevice> devices = <FlutterDevice>[];
+
+    setUp(() {
+      devices.add(new FlutterDevice(new MockDevice(),
+          previewDart2: false, trackWidgetCreation: false));
+    });
+
+    testUsingContext('no setup', () async {
+      expect((await new HotRunner(devices).restart(fullRestart: true)).isOk,
+          true);
+    });
+
+    testUsingContext('setup function succeeds', () async {
+      expect((await new HotRunner(devices).restart(fullRestart: true)).isOk,
+          true);
+    }, overrides: <Type, Generator>{
+      HotRunnerConfig: () => new TestHotRunnerConfig(true),
```

Comment:
Do you think it make sense to make `successfulSetup` a named parameter for the constructor so it is clear what `true` and `false` parameter mean here?